tests/new_upgrades/test_role.py

```diff
@@ -0,0 +1,259 @@
+"""Test for Role related Upgrade Scenario's
+
+:Requirement: UpgradedSatellite
+
+:CaseAutomation: Automated
+
+:CaseComponent: UsersRoles
+
+:Team: Endeavour
+
+:CaseImportance: High
+
+"""
+
+from box import Box
+import pytest
+
+from robottelo.utils.shared_resource import SharedResource
+
+
+@pytest.fixture
+def default_role_permission_setup(search_upgrade_shared_satellite, upgrade_action):
+    """New permission is added to Default Role
+
+    :steps: New permission is added to existing 'Default role'
+
+    :expectedresults: Permission is added to existing 'Default role'.
+    """
+    target_sat = search_upgrade_shared_satellite
+    with SharedResource(target_sat.hostname, upgrade_action, target_sat=target_sat) as sat_upgrade:
+        default_role = target_sat.api.Role().search(query={'search': 'name="Default role"'})[0]
+        subnet_filter = target_sat.api.Filter(
+            permission=target_sat.api.Permission().search(
+                filters={'name': 'view_subnets'}, query={'search': 'resource_type="Subnet"'}
+            ),
+            role=default_role,
+        ).create()
+        assert subnet_filter.id in [filt.id for filt in default_role.read().filters]
+        test_data = Box(
+            {
+                'satellite': target_sat,
+            }
+        )
+        sat_upgrade.ready()
+        target_sat._session = None
+        yield test_data
+
+
+@pytest.mark.search_upgrades
+def test_default_role_added_permission(default_role_permission_setup):
+    """The new permission in 'Default role' is intact post upgrade
+
+    :id: e4a-96b3-4033-b562-3130fc43a4bc
+
+    :expectedresults: The added permission in existing 'Default role' is
+        intact post upgrade
+    """
+    target_sat = default_role_permission_setup.satellite
+    default_role = target_sat.api.Role().search(query={'search': 'name="Default role"'})[0]
+    subnet_filter = target_sat.api.Filter().search(
+        query={'search': f'role_id={default_role.id} and permission="view_subnets"'}
+    )
+    assert subnet_filter
+    # Teardown
+    subnet_filter[0].delete()
+
+
+@pytest.fixture
+def default_role_permission_with_filter_setup(search_upgrade_shared_satellite, upgrade_action):
```

Comment:
nitpick, I'd move this up for the module to start with fixtures before getting to the tests